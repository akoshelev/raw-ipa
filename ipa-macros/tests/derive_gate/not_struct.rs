use ipa_macros::Gate;

// derive Gate supports struct only
#[derive(Gate)]
enum MyGate {}

fn main() {}
