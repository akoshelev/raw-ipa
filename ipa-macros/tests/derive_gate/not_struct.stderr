error: proc-macro derive panicked
 --> tests/derive_gate/not_struct.rs:4:10
  |
4 | #[derive(Gate)]
  |          ^^^^
  |
  = help: message: derive Gate expects a struct
